src/core/component.ts

```diff
@@ -34,26 +31,25 @@ abstract class ComponentBase<P> extends React.PureComponent<P, IState> {
 
   protected readonly _templateProps: ITemplateMeta[] = [];
 
-  private readonly _templateHelper: TemplateHelper;
   private readonly _optionsManager: OptionsManager;
 
   constructor(props: P) {
     super(props);
     this._prepareProps = this._prepareProps.bind(this);
+    this._updateState = this._updateState.bind(this);
 
     this.state = {
       templates: {}
     };
 
     this._optionsManager = new OptionsManager((name) => this.props[name]);
-    this._templateHelper = new TemplateHelper(this);
   }
 
   public componentWillUpdate(nextProps: P) {
-    const preparedProps = this._prepareProps(nextProps);
+    const preparedProps = this._prepareProps(nextProps, this._defaults, this._templateProps);
```

Comment:
As the `_prepareProps` is an instance method, is in necessary to pass `_defaults` and `_templateProps` as params?